src/engine/threaded_engine.h

```diff
@@ -293,6 +299,24 @@ class ThreadedEngine : public Engine {
     finished_cv_.notify_all();
   }
 
+  /*! \brief Return default OMP thread count. Currently, this is whatever OMP shows as number
+   * of procs
+   */
+  static int DefaultOMPThreadsPerWorker() {
+#ifdef _OPENMP
+    // If environment variable is set and it's not empty, then use omp_get_max_threads()
+    // (Check environment directly, since OMP_NUM_THREADS mnay have odd formatting (i.e. 3, 2"))
+    // Taking max including omp_get_max_threads() in case this implementation of OMP accounts for
+    // hyperthreading
+    const char *s = getenv("OMP_NUM_THREADS");
+    const int max_threads = s && *s ? omp_get_max_threads() : std::max(omp_get_max_threads(),
+                                                                       omp_get_num_procs());
```

Comment:
Still don't quite understand the max here.
If OMP_NUM_THREADS is not set, what does omp_get_max_threads and omp_get_num_procs each return?
We should use the number of real (i.e. without hyperthreading) cores.

Reply:
procs is without hyperthreading. threads() is whatever the openmp decides is the best max (on my machine, it is same as procs()).

Reply:
I will just change to  omp_get_max_threads(). I am working on managing this directly for nested situations soon anyway.